Applications/ConsumerApi/test/ConsumerApi.Tests.Performance/src/libs/backbone-client/fluent-client.ts

```diff
@@ -0,0 +1,80 @@
+// @ts-expect-error: k6 uses links to packages, which typescript cannot lint.
+import { Httpx } from "https://jslib.k6.io/httpx/0.1.0/index.js";
+
+export class FluentClient {
+    private readonly client: Httpx;
+    private endpoint!: string;
+    private headers!: Record<string, string | number | string[]>;
+
+    public constructor(client: Httpx) {
+        this.client = client;
+    }
+
+    public setEndpoint(endpoint: string): this {
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    public setHeaders(headers: Record<string, string | number | string[]>): this {
+        this.headers = headers;
+        return this;
+    }
+
+    public authenticate(token: string): this {
+        this.headers = { ...this.headers, Authorization: token };
+        return this;
+    }
+
+    public asJson(): this {
```

Comment:
Shouldn't this rather be called seomething like "withJsonBody"? "asJson" looks as if the method would return a json representation of the FluentClient.